tedana/reporting/html_report.py

```diff
@@ -138,44 +143,69 @@ def _update_template_bokeh(bokeh_id, info_table, about, prefix, references, boke
     -------
     HTMLReport : an instance of a populated HTML report
     """
-    resource_path = Path(__file__).resolve().parent.joinpath("data", "html")
-
     # Initial carpet plot (default one)
     initial_carpet = f"./figures/{prefix}carpet_optcom.svg"
 
+    # Get the figures directory - relative to the directory containing the references file
+    base_dir = os.path.dirname(references)
+    figures_dir = os.path.join(base_dir, "figures")
+
+    # List all files in the figures directory
+    files_in_figures = os.listdir(figures_dir)
+
     # Adaptive mask image
-    adaptive_mask = f"./figures/{prefix}adaptive_mask.svg"
+    adaptive_mask_filename = f"{prefix}adaptive_mask.svg"
+    adaptive_mask = f"./figures/{adaptive_mask_filename}"
+    adaptive_mask_exists = adaptive_mask_filename in files_in_figures
+    LGR.info(
+        f"Checking for adaptive mask: {adaptive_mask_filename}, exists: {adaptive_mask_exists}"
+    )
 
-    # T2* and S0 images
-    t2star_brain = f"./figures/{prefix}t2star_brain.svg"
-    t2star_histogram = f"./figures/{prefix}t2star_histogram.svg"
-    s0_brain = f"./figures/{prefix}s0_brain.svg"
-    s0_histogram = f"./figures/{prefix}s0_histogram.svg"
-    rmse_brain = f"./figures/{prefix}rmse_brain.svg"
-    rmse_timeseries = f"./figures/{prefix}rmse_timeseries.svg"
+    # Check if essential T2* and S0 images exist
+    t2star_brain_filename = f"{prefix}t2star_brain.svg"
+    t2star_histogram_filename = f"{prefix}t2star_histogram.svg"
+    s0_brain_filename = f"{prefix}s0_brain.svg"
+    s0_histogram_filename = f"{prefix}s0_histogram.svg"
+    rmse_brain_filename = f"{prefix}rmse_brain.svg"
+    rmse_timeseries_filename = f"{prefix}rmse_timeseries.svg"
+
+    # T2* and S0 images for HTML
+    t2star_brain = f"./figures/{t2star_brain_filename}"
+    t2star_histogram = f"./figures/{t2star_histogram_filename}"
+    s0_brain = f"./figures/{s0_brain_filename}"
+    s0_histogram = f"./figures/{s0_histogram_filename}"
+    rmse_brain = f"./figures/{rmse_brain_filename}"
+    rmse_timeseries = f"./figures/{rmse_timeseries_filename}"
+
+    t2s0_exists = (
```

Comment:
If you provide a T2* image to the command-line, the T2* file will exist, but the S0 image will not. Do we not generate the T2* brain and histogram images in that scenario?